plugins/json_importer.cc

```diff
@@ -0,0 +1,239 @@
+                        return;
+                    }
+                    else if(val.isBool()) {
+                        outputRow.columns.emplace_back(ColumnName(id), val.asBool(), zeroTs); 
+                    }
+                    else if(val.isInt()) {
+                        outputRow.columns.emplace_back(ColumnName(id), val.asInt(), zeroTs); 
+                    }
+                    else if(val.isDouble()) {
+                        outputRow.columns.emplace_back(ColumnName(id), val.asDouble(), zeroTs); 
+                    }
+                    else if(val.isString()) {
+                        outputRow.columns.emplace_back(ColumnName(id), val.asString(), zeroTs); 
+                    }
+                    else if(val.isArray()) {
+                        // is it only atomic types?
+                        bool onlyAtomic = true;
+                        for(int i=0; i<val.size(); i++) {
+                            if(val[i].isArray() || val[i].isObject()) {
+                                onlyAtomic = false;
+                                break;
+                            }
+                        }
+
+                        if(onlyAtomic) {
+                            for(int i=0; i<val.size(); i++) {
+                                string key;
+                                if(val[i].isString())      key = val[i].asString();
+                                else if(val[i].isBool())   key = val[i].asBool() ? "true" : false;
+                                else if(val[i].isDouble()) key = boost::lexical_cast<string>(val[i].asDouble());
+                                else if(val[i].isInt())    key = boost::lexical_cast<string>(val[i].asInt());
+                                else                       key = val[i].toString();
+                                emplaceCol(id + '_' + key, Json::Value(true));
+                            }
+                        }
+                        else {
+                            auto str = val.toString();
+                            if(str.substr(str.size()-1) == "\n")
+                                str = str.substr(0, str.size() -1);
+                            outputRow.columns.emplace_back(ColumnName(id), std::move(str), zeroTs); 
+                        }
+                    }
+                    else if(val.isObject()) {
+                        for (const std::string & sub_id : val.getMemberNames()) {
+                            emplaceCol(id + '_' + sub_id, val[sub_id]);
+                        }
+                    }
+                };
+
+            for (const std::string & id : root.getMemberNames()) {
+                emplaceCol(id, root[id]);
+            }
+            
+            recordedLines++;
+
+            std::unique_lock<std::mutex> guard(recordMutex);
+            output->recordRow(outputRow.rowName, outputRow.columns);
+            return true;
+        };
+
+
+        forEachLineStr(runProcConf.dataFileUrl.toString(),
```

Comment:
The csv importer was changed to use forEachLineBlock for better performance.  Consider using the same here.